apiserver/facades/client/application/application.go

```diff
@@ -230,12 +246,18 @@ func newFacadeBase(ctx facade.Context) (*APIBase, error) {
 
 	resources := ctx.Resources()
 
+	leadershipReader, err := ctx.LeadershipReader(ctx.State().ModelUUID())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
 	return NewAPIBase(
 		&stateShim{ctx.State()},
 		storageAccess,
 		ctx.Auth(),
 		blockChecker,
-		facadeModel,
+		&modelShim{facadeModel},
```

Comment:
It isn't clear here why there is a shim. Comment maybe?

Reply:
We need a shim when we need to use a method that returns an interface as opposed to a state struct.  I'll add a comment